tests/unit/test_base_collection.py

```diff
@@ -0,0 +1,133 @@
+from unittest import TestCase
+
+import responses
+
+from tamr_unify_client import Client
+from tamr_unify_client.auth import UsernamePasswordAuth
+
+
+class DeleteByResourceID(TestCase):
+    def setUp(self):
+        auth = UsernamePasswordAuth("username", "password")
+        self.unify = Client(auth)
+
+    @responses.activate
+    def test_dataset_collection(self):
+        ds_url = self.url_prefix + "datasets/115"
+
+        responses.add(responses.DELETE, ds_url, status=204)
+
+        response = self.unify.datasets.delete_by_resource_id("115")
+        self.assertEqual(response.status_code, 204)
+
+    @responses.activate
+    def test_attr_config_collection(self):
```

Comment:
BaseCollection needs to know nothing about AttributeConfigurationCollection; this test needs to move out of this file and into an AttributeConfigurationCollection test file.